docs/source/02_architecture.md

```diff
@@ -0,0 +1,123 @@
+# Architecture {#architecture}
+
+ADAM is architected as an extensible, parallel framework for working with both
+aligned and unaligned genomic data using [Apache
+Spark](https://spark.apache.org). Unlike traditional genomics tools, ADAM is
+built as a modular stack, where a set of schemas define and provide a narrow
+waist. This stack architecture enables the support of a wide range of data
+formats and optimized query patterns without changing the data structures and
+query patterns that users are programming against.
+
+## The ADAM Stack Model {#stack-model}
+
+The stack model that ADAM is based upon was introduced in [@massie13] and
+further refined in [@nothaft15], and is depicted in the figure below. This
+stack model seperates computational patterns from the data model, and the
+data model from the serialized representation of the data on disk. This
+enables developers to write queries that will run seamlessly on both a single
+node, or on a distributed cluster, on legacy genomics data files or on
+data stored in a high performance columnar storage format, on sorted or
+unsorted data, without making any modifications to their query. Additionally,
```

Comment:
How about

This enables developers to write queries that run seamlessly on a single
node or on a distributed cluster, on legacy genomics data files or on
data stored in a high performance columnar storage format, and on sorted or
unsorted data, without making any modifications to their query.